pkg/webhooks/github/actions/common.go

```diff
@@ -186,23 +197,23 @@ func (w *WebhookActions) ProcessReleaseEvent(ctx context.Context, payload *ghweb
 	}
 }
 
-func (w *WebhookActions) ProcessPullRequestEvent(ctx context.Context, payload *ghwebhooks.PullRequestPayload) {
+func (w *WebhookActions) ProcessPullRequestEvent(ctx context.Context, payload *github.PullRequestEvent) {
 	ctx, cancel := context.WithTimeout(ctx, constants.WebhookHandleTimeout)
 	defer cancel()
 	g, _ := errgroup.WithContext(ctx)
 
 	for _, a := range w.dp {
 		g.Go(func() error {
-			if payload.Action != "opened" || payload.Repository.Name != "docs" {
+			if pointer.SafeDeref(payload.Action) != "opened" || pointer.SafeDeref(payload.Repo.Name) != "docs" {
 				return nil
 			}
 			params := &docsPreviewCommentParams{
-				PullRequestNumber: int(payload.PullRequest.Number),
+				PullRequestNumber: int(pointer.SafeDeref(payload.PullRequest.Number)),
 			}
 
 			err := a.AddDocsPreviewComment(ctx, params)
 			if err != nil {
-				w.logger.Error("error adding docs preview comment to docs", "repo", payload.Repository.Name, "pull_request", params.PullRequestNumber, "error", err)
+				w.logger.Error("error adding docs preview comment to docs", "repo", *payload.Repo.Name, "pull_request", params.PullRequestNumber, "error", err)
```

Comment:
not that this should happen in practice, but in case `payload.Repo.Name` is actually `nil`, it is likely to produce an error within this request.
Should we add a `var`-block above with multiple `pointer.SafeDeref`?

Reply:
The nil validations are really painful with the new event definitions... I am not sure how to deal with it. Also if there was an empty string the behavior of the robot would turn into something completely unforeseeable. Maybe there's smart solution with JSON marshaling and unmarshaling + defining a struct linter?

Reply:
Applied your suggestion, but it's a bit harder to review now.